salt/modules/rpmbuild.py

```diff
@@ -265,13 +271,117 @@ def build(runas,
     return ret
 
 
-def make_repo(repodir, keyid=None, env=None):
+def make_repo(repodir, keyid=None, env=None, use_passphrase=False, gnupghome='/etc/salt/gpgkeys', runas='root'):
     '''
     Given the repodir, create a yum repository out of the rpms therein
 
     CLI Example::
 
         salt '*' pkgbuild.make_repo /var/www/html/
+
+    repodir
+        The directory to find packages that will be in the repository
+
+    keyid
+        Optional Key ID to use in signing repository
+
+    env
+        Optional dictionary of environment variables to be utlilized in creating the repository.
+
+        .. versionadded:: 2016.3.0
+
+    use_passphrase
+        Whether to use a passphrase with the signing key.  Passphrase is received from pillar.
+
+    gnupghome
+        Specify the location where GPG related files are stored.
+
+    runas
+        User to create the repository as
     '''
-    cmd = 'createrepo {0}'.format(repodir)
-    __salt__['cmd.run'](cmd)
+    # ensure all product owned by runas
+    __salt__['cmd.run']('chown {0}:{0} -R {1}'.format(runas, repodir))
+
+    SIGN_PROMPT_RE = re.compile(r'Enter pass phrase: ', re.M)
+
+    local_keyid = None
+    define_gpg_name = ''
+    local_uids = None
+
+    if keyid is not None:
+        ## import_keys
+        pkg_pub_key_file = '{0}/{1}'.format(gnupghome, __salt__['pillar.get']('gpg_pkg_pub_keyname'))
```

Comment:
I shall add documentation stating use of keys in pillar data and error-out if not present.  The documentation for salt-pack which shows use of the public / private keys is planned for the next release, as part of the update for Ubuntu 16.04 LTS.